common/persistence/visibility/store/elasticsearch/query_interceptors.go

```diff
@@ -120,19 +148,23 @@ func (ni *nameInterceptor) Name(name string, usage query.FieldNameUsage) (string
 }
 
 func (vi *valuesInterceptor) Values(fieldName string, values ...interface{}) ([]interface{}, error) {
-	fieldType, err := vi.searchAttributesTypeMap.GetType(fieldName)
+	name := fieldName
+	if _, exists := vi.nameInterceptor.fieldTransformations[fieldName]; exists {
+		name = vi.nameInterceptor.fieldTransformations[fieldName].newField
+	}
+
+	fieldType, err := vi.searchAttributesTypeMap.GetType(name)
 	if err != nil {
 		return nil, query.NewConverterError("invalid search attribute: %s", fieldName)
 	}
 
-	name := fieldName
-	if searchattribute.IsMappable(fieldName) {
+	if searchattribute.IsMappable(name) {
 		mapper, err := vi.searchAttributesMapperProvider.GetMapper(vi.namespace)
 		if err != nil {
 			return nil, err
 		}
 		if mapper != nil {
-			name, err = mapper.GetAlias(fieldName, vi.namespace.String())
+			name, err = mapper.GetAlias(name, vi.namespace.String())
```

Comment:
This should be `fieldName`, not alias.

Reply:
It's synonymous with `fieldName` unless we remap:

```
	if _, exists := vi.nameInterceptor.fieldTransformations[fieldName]; exists {
		name = vi.nameInterceptor.fieldTransformations[fieldName].newField
	}
```